addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go

```diff
@@ -321,72 +321,100 @@ func (h *Helper) HandleExistingClusterBootstrap(clusterBootstrap *runtanzuv1alph
 		clusterBootstrap.Spec.Kapp,
 	}, clusterBootstrap.Spec.AdditionalPackages...)
 
-	if clusterBootstrap.Annotations != nil {
-		if annotationValue, annotationExist := clusterBootstrap.Annotations[constants.AddCBMissingFieldsAnnotationKey]; annotationExist {
-			tkrName = annotationValue
-			clusterBootstrapTemplateName := annotationValue // TanzuKubernetesRelease and ClusterBootstrapTemplate share the same name
-			clusterBootstrapTemplate := &runtanzuv1alpha3.ClusterBootstrapTemplate{}
-			// Get the ClusterBootstrapTemplate mentioned in the annotation under system namespace
-			if err := h.K8sClient.Get(h.Ctx, client.ObjectKey{Namespace: systemNamespace, Name: clusterBootstrapTemplateName}, clusterBootstrapTemplate); err != nil {
-				return nil, err
-			}
+	if annotationValue, annotationExist := clusterBootstrap.Annotations[constants.AddCBMissingFieldsAnnotationKey]; annotationExist {
```

Comment:
**Question**: Would Annotations be nil anytime ? Should we handle it ?

Reply:
The statment 
```
if annotationValue, annotationExist := clusterBootstrap.Annotations[constants.AddCBMissingFieldsAnnotationKey]; annotationExist
``
is special in golang and if annotations = nil, then annotationExist == ... returns nil. It won't error out. 
you can run this code to verify: 

```type ClusterBootstrap struct {
	Annotations map[string]string
}

func main() {
	clusterBootstrap := ClusterBootstrap{}
	if annotationValue, annotationExist := clusterBootstrap.Annotations["somefield"]; annotationExist {
		println(annotationValue)
	} else {
		println("Annotatinons == nil but we! good it works")
	}
}
```

Reply:
Sounds good. Just wanted to confirm.